src/gmpints.c

```diff
@@ -375,6 +375,31 @@ Obj ObjInt_UInt( UInt i )
   }
 }
 
+Obj ObjInt_Int8( Int8 i )
+{
+  int neg = 0;
+  if(i < 0) {
+    neg = 1;
+    i = -i;
+  }
```

Comment:
If you look at git head (I added more code), the overhead isn't quite as bad, because (and I didn't remember this) on a 64-bit machine Int == Int8, so this method is only used on 32-bit machines, on values between 32-bits and 64-bits.